index.rst

```diff
@@ -48,12 +48,336 @@
+   To support this, the ticket number will become a new required input to the Gen2 to Gen3 conversion tooling.
+
+
+Filesystem locations
+====================
+
+The main shared data repository for all instruments at NCSA will have a public repository root of ``/datasets/repo``, which will be a symlink to a directory of the form ``/datasets/repo_<YYYYMMDD>``.
+These directories will each contain a ``butler.repo`` yaml that points to the appropriate database (with a one-to-one correspondance between databases or database schemas and ``repo_<YYYYMMDD>`` directories).
+
+The default (POSIX) datastore will write datasets with templates that begin with the ``RUN`` name, resulting in e.g. the datasets of per-instrument ``RUN`` collections landing in ``/datasets/repo_<YYYYMMDD>/<instrument>/`` and per-user ``RUN`` collections landing in ``/datasets/repo_<YYYYMMDD>/u/<user>``.
+Users are discouraged from inspecting these directories (as this will be at least quite different in the IDF or other future cloud-based datastores), and _strongly_ discouraged from modifying them in any way other than via middleware tools.
+In many cases, write access will actually be prohibited (see :ref:`access-controls`).
+
+When migrations are necessary due to changes in the repository format (something that is _always_ preceded by an RFC with explicit CCB approval), a new ``repo_<YYYYMMDD>`` directory and database/schema pair will be created, and files will shared via hard links until/unless the old repository is retired.
+
+.. note::
+
+   **TODO**: are hard links viable here from a sysadmin/GPFS perspective?
+   They certainly would make things easier.
+
+We will also designate two other non-repository subdirectories of ``/datasets`` for specific roles:
+
+ - ``/datasets/external`` holds files produced by other projects or surveys that may be of use to multiple users but does not fit into the LSST data model.  This includes the original versions of reference catalogs (e.g. Gaia DR2), truth catalogs (e.g. from DESC DC2), dust maps, etc.  Each subdirectory should have a descriptive README, and no files should be put in ``datasets/external`` itself.
+
+ - ``/datasets/testing`` holds git LFS repositories that are used in CI and rarely change.  These are provided for convenience, and should be updated when their git master branches are; there will be no attempt to make old versions available.
+
+Finally, we propose that all raw ingestion into the shared repository be done with symbolic links to read-only filesystems - the existing ``/lsstdata`` for Rubin Observatory data, and a new ``/external-raw`` filesystem for raw data from other instruments.
+
+.. note ::
+
+   **TODO**: get feedback from RHL and NCSA on staging directory substructure, and the workflow for ingestion.
```

Comment:
I'm not sure exactly what details you're looking for here.  Currently raw images get transferred to NCSA via a couple different mechanisms (plain rsync for BOT and CCS-written images or DBB Handoff Buffer Manager service currently using bbcp).  The DBB Endpoint Buffer Manager service moves the file to a permanent read-only location in /lsstdata/offline (currently still teststand, but expected to transition to dbb subdir).  It also calls Gen2 and Gen3 ingestions (via python APIs) on each new file.  It keeps track of timings and success/failures in database tables.  Reingestion of files should, under normal circumstances, be done by "asking" the DBB BM to ingest the file again so that the database information stays valid.    We were interested in trying out the "in-place" ingestion instead of making a bunch of softlinks, but if users want to be able to manually walk filesystems and see butler directories and filenames we can continue doing the softlinks.

Reply:
I'm actively working on the "direct" transfer mode that will put the full URI in the datastore.

Reply:
When to run define-visits is still an interesting question in the absence of a header telling us how many observations are  in the visit.

Reply:
I suppose it's not so much a matter of feedback as just asking for someone else to write something here; I don't know enough to be the one to do it.  I'll update the TODO note to reflect that, and we can work out later how that gets done.

And I have no problem with using in-place ingestion instead of symlinks for raws; I think I brought them up here only to emphasize that I wanted to invert the permissions/filesystems relationship: instead of `/datasets` being read-only (to devs) with symlinks there to storage on `/project`, I'd rather most of `/datasets` be read-write, with symlinks to the really important read-only-to-almost-everything storage on `/lsstdata` (and of course read-only ingest would accomplish that, too).

Reply:
I've dropped this to-do note from the RFC (I think it's fine for the organization of `/lsstdata` to be out of scope for this document), and have proposed using in-place transfers in the text and a few new tables.